doc/030_preparing_a_new_repo.rst

```diff
@@ -0,0 +1,250 @@
+   option ``--password-command`` or the environment variable
+   ``RUSTIC_PASSWORD_COMMAND``
+   
+The ``init`` command has an option called ``--set-version`` which can
+be used to explicitely set the version for the new repository.
+
+The below table shows which rustic version is required to use a certain
+repository version and shows new features introduced by the repository format.
+
++--------------------+------------------------+---------------------+
+| Repository version | Minimum rustic version | Major new features  |
++====================+========================+=====================+
+| ``1``              | any version            |                     |
++--------------------+------------------------+---------------------+
+| ``2``              | >= 0.2.0               | Compression support |
++--------------------+------------------------+---------------------+
+
+Moreover, there are different options which can be set when initializing
+a repository:
+
+Options to specify the target pack size:
+
+ * ``--set-treepack-size``, ``--set-datapack-size`` specify the default target pack
+   size for tree and data pack files. Arguments can given using TODO
+   For example, valid sizes are "4048kiB", "2MB", "30MiB", etc. 
+   If not specified, the default is 4 MiB for tree packs and 32 MiB for data packs.
+
+ * ``--set-treepack-growfactor``, ``--set-datapack-growfactor`` specify how much the 
+   target pack size should be increased per square root of the total pack size in bytes
+   of the given type. This equals to 32kiB per square root of the total pack size in GiB.
+
+
+
+Note that larger pack sizes have advantages, especially for large repository or remote 
+repositories. They lead to less packs in the repository and transfer larger datasets 
+to the repository which can increase the throughput.
+But there are also disadvantages. Rustic keeps the whole pack in memory before writing it 
+to the backend. As writes are parallelized, multiple packs are kept. So larger pack sizes
+increase the memory usage of the ``backup`` command. Moreover larger pack sizes lead to
+increased repack rates during ``prune`` or ``forget -- prune``.
+
+Local
+*****
+
+In order to create a repository at ``/srv/rustic-repo``, run the following
+command and enter the same password twice:
+
+.. code-block:: console
+
+    $ rustic init -r /srv/rustic-repo
+    enter password for new repository:
+    created rustic repository 085b3c76b9 at /srv/rustic-repo
+
+.. warning::
+
+   Remembering your password is important! If you lose it, you won't be
+   able to access data stored in the repository.
+
+.. warning::
+
+   On Linux, storing the backup repository on a CIFS (SMB) share is not
+   recommended due to compatibility issues. Either use another backend
+   or set the environment variable `GODEBUG` to `asyncpreemptoff=1`.
+   Refer to GitHub issue `#2659 <https://github.com/rustic/rustic/issues/2659>`_ for further explanations.
```

Comment:
It's possible that this is not relevant to rustic, but I not checked. Also, correct url is https://github.com/restic/restic/issues/2659